iavl/visualize.py

```diff
@@ -0,0 +1,49 @@
+import binascii
+from typing import List
+
+from graphviz import Digraph
+from hexbytes import HexBytes
+
+from .utils import Node, decode_node, node_key, store_prefix
+
+
+def label(node: Node):
+    s = binascii.hexlify(node.key).decode()
+    if len(s) > 10:
+        s = s[:4] + "..." + s[-4:]
+    s += f"({node.version})"
+    if node.is_leaf():
+        s += f"\nvalue len: {len(node.value)}"
+    return s
+
+
+def visualize_iavl(db, store: str, root_hash: bytes, version: int) -> Digraph:
+    g = Digraph(comment="IAVL Tree")
+    prefix = store_prefix(store)
+
+    def get_node(hash: bytes) -> Node:
+        n, _ = decode_node(db.get(prefix + node_key(hash)))
+        return n
+
+    def vis_node(hash: bytes, n: Node):
+        style = "solid" if n.version == version else "filled"
+        g.node(HexBytes(hash).hex(), label=label(node), style=style)
+
+    stack: List[bytes] = [root_hash]
+    while stack:
+        hash = stack.pop()
```

Comment:
do we need check already expanded like [isinstance(hash, tuple)](https://github.com/crypto-com/python-iavl/blob/c0cea446ee9395656264a3cbdedc4b9dcad4bfc8/iavl/utils.py#L326)

Reply:
no, that's for post-order traversal, we only do preorder here.